src/language-js/print/object.js

```diff
@@ -237,4 +232,13 @@ function printObject(path, options, print) {
   return group(content, { shouldBreak });
 }
 
+function shouldHugParameter(node, name, number) {
+  return (
+    ((number === 0 && (name === "params" || name === "parameters")) ||
+      name === "this" ||
+      name === "rest") &&
+    shouldHugFunctionParameters(node)
```

Comment:
If function parameters has `.this` and normal parameters, won't this pass on second parameter?

Reply:
Use `getFunctionParameters().indexOf`?

Reply:
`shouldHugFunctionParameters` checks that there is only one parameter, but yes, doing what you propose would be easier to follow.

Reply:
If `shouldHugFunctionParameters` only pass on one parameter. The number check is redundant. Only need make sure it's a parameter.

Reply:
I'm a little bit confused, the "node" is a function, right?  It can't be another function's parameter, how `name` can be "parameters"? The "name" is key of child?

Reply:
Ah yes, this is different from callback of `AstPath#each`, the name is key of node there.

Reply:
Actually, `getFunctionParameters().indexOf` looks clumsy because I need to pass a node to it, and to get this node I have to use `name` and `number` anyway. So I'm going to simply remove the number check.

Reply:
I also renamed `shouldHugFunctionParameters` to `shouldHugTheOnlyFunctionParameter` to make it clear that it checks that there is only one parameter.